core/shared/src/test/scala/cats/parse/StringInterpolatorTest.scala

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright (c) 2021 Typelevel
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of
+ * this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+package cats.parse
+
+import StringInterpolation._
+
+class StringInterpolatorTest extends munit.ScalaCheckSuite {
+
+  test("parser string interpolator") {
+    val foo: Parser[Boolean] = Parser.string("spam").as(true) | Parser.string("eggs").as(false)
+    val bar: Parser[BigInt] = Numbers.bigInt
+    val quux: Parser[Int] = Parser.string("quux").as(42)
+
```

Comment:
can we exercise with 0 args, 1 arg up to say 3 (to exercise non-tuples)
so, I guess `val x: Parser[Unit] = parser"foo"` should work. `val x: Parser[A] = parser"foo$aParser"` and then at 2 we should have tuples?

or do we require 2 or more (or 1 if we wrap with tuple1)?

Reply:
Wrote test to exercise these cases. Here's how it currently works:

* empty string: compile error
* non-empty string with no interpolated parsers: `Parser[Unit]` equivalent to `Parser.string("...")`
* string with single interpolated `Parser[A]`: `Parser[A]`
* string with n interpolated `Parser[{A, B, C...}]`: `Parser[(A, B, C...)]`